lib/conjur/api/ldapsync.rb

```diff
@@ -70,22 +65,52 @@ def ldap_sync_show_profile(profile, options = {})
     #
     # @param [Hash] profile a hash containing the LDAP sync configuration
     # @param [Hash] options reserved
-    def ldap_sync_update_profile(profile, options = {})
+    def ldap_sync_update_profile(profile_name, profile, options = {})
       options[:json_config] = profile.to_json
-      resp = RestClient::Resource.new(Conjur.configuration.appliance_url, credentials)['ldap-sync']['config'].put(options.to_json, :content_type => 'application/json')
+      resp = RestClient::Resource.new(Conjur.configuration.appliance_url, credentials)['ldap-sync']['config'][profile_name].put(options.to_json, :content_type => 'application/json')
       JSON.parse(resp.body)
     end
 
+    # @api private
+    # Search using an LDAP sync profile
+    #
+    # @param [String] profile name
+    # @param [Hash] options reserved
+    def ldap_sync_search(profile, options = {})
+      headers = credentials.dup.tap {|h|
+        h[:headers][:accept] = 'text/event-stream'
+      }
+
+      response = RestClient::Resource.new(Conjur.configuration.appliance_url, headers)['ldap-sync']['search'].post(options.merge(:config_name => profile))
+      JSON.parse(get_json("groups", response)).merge('events' => find_log_events(response))
+    end
+
     # @!endgroup
     
     private
-    def find_policy_event(response)
-      response.body.lines.find {|l| l =~ /^data: {"policy":/}.try(:[], 6..-1)
+    def get_json(ok_key, response)
+      if response.headers[:content_type] == 'text/event-stream'
+        find_ok_event(ok_key, response) || find_error_events(response)
+      else
+        %Q({"error": {"message": "Unexpected response from server: #{response.body}"}})
+      end    
+    end
+
+    def find_ok_event(key, response)
+      response.body.lines.find {|l| l =~ %r(^data: {"#{key}":) }.try(:[], 6..-1)
+    end
+
+    def find_log_events(response)
+      find_events(response, 'log').collect { |e| JSON.parse(e)['log'] }
     end
 
     def find_error_events(response)
+      find_events(response, "error").join("\n")
       response.body.lines.collect {|l| l.match(/^data: ({"error":.*)/).try(:[], 1)}.compact.join("\n")
```

Comment:
Can you get rid of this line?  It looks like you're performing the same filter twice:
```
def find_error_events(response)
  find_events(response, "error").join("\n")
end
```